tests/FSharp.Data.Tests/Http.fs

```diff
@@ -8,22 +8,85 @@ open System.Net
 open FSharp.Data
 open FSharp.Data.HttpRequestHeaders
 open System.Text
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Builder
+open Microsoft.AspNetCore.Http
+open System.Net.NetworkInformation
+
+type ITestHttpServer =
+    inherit IDisposable
+    abstract member BaseAddress: string
+    abstract member WorkerTask: Task
+
+let startHttpLocalServer() =
+    let app = WebApplication.CreateBuilder().Build()
+    app.Map("/{status}", (fun (ctx: HttpContext) ->
+        async {
+            match ctx.Request.RouteValues.TryGetValue("status") with
+            | true, (:? string as status) ->
+                let status = status |> int
+
+                match ctx.Request.Query.TryGetValue("sleep") with
+                | true, values when values.Count = 1 ->
+                    let value = values[0] |> int
+                    do! Async.Sleep value
+                | _ -> ()
+
+                if ctx.Request.Body <> null then
+                    let buffer = Array.create 8192 (byte 0)
+                    let mutable read = -1
+                    while read <> 0 do
+                        let! x = ctx.Request.Body.ReadAsync(buffer, 0, 8192) |> Async.AwaitTask
+                        read <- x
+
+                Results.StatusCode(status).ExecuteAsync(ctx)
+                |> Async.AwaitTask
+                |> ignore
+            | _ -> failwith "Unexpected request."
+                
+        } |> Async.StartAsTask :> Task
+        )) |> ignore
+
+    let freePort =
+        let mutable port = 55555 // base listener port for the tests
+        while
+            IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpListeners()
+            |> Array.map (fun x -> x.Port)
+            |> Array.contains port do
+                port <- port + 1
+        port
+
+    let baseAddress = $"http://localhost:{freePort}"
+
+    let workerTask = app.RunAsync(baseAddress)
+    printfn $"Started local http server with address {baseAddress}"
+
+    { new ITestHttpServer with
+        member this.Dispose() =
+            app.StopAsync() |> Async.AwaitTask |> ignore
+            printfn $"Stopped local http server with address {baseAddress}"
+        member this.WorkerTask = workerTask
+        member this.BaseAddress = baseAddress }
 
 [<Test>]
 let ``Don't throw exceptions on http error`` () =
-    let response = Http.Request("http://httpstat.us/401", silentHttpErrors = true)
-    response.StatusCode |> should equal 401
+    let localServer = startHttpLocalServer()
+    using localServer <| fun _ ->
```

Comment:
Any reason not to do this?

```fsharp
use localServer = startHttplocalServer()
//...
```

Reply:
I started with an implementation where the `IDisposable returned by `startHttpLocalServer()` didn't contain anything useful beside `Dispose()` and I was afraid `Dispose()` would be called too early if the value wasn't otherwise used. (I'm not sure when it goes out of scope in this situation)

This is less true now that the `BaseAddress` is used to create the http request.

Do you want me to change it? (and if you know about when `Dispose()` is called with seemingly unused `use` bindings, I'd be interested to know :))

Reply:
It will only get called when the whole function is out of scope, see here: https://sharplab.io/#v2:DYLgZgzgPg9gDgUwHYAIDKBPCAXBBbAWACh5l0tc8A6AFQAsAnBAQwBMBLJAc1uYgGsIxYtgyIUASRoIcACWzY4aBAwBuKlAF5iKXSk50V7bJIAi7CHBgRmAI2AIdeuzgbMAxibz5bGgEJ8CACCrKxMEBAgKK6cXE66LthunijeeL4MKADqMAz8KjR8/FGFAsJEDia2zAxgKAAUAJRa8SgA3ihICADuktJyCkoq6pndxnSteqk+Gth0FlTmltYITS1EU5t6cAyc2GCoAEQYCAjYh5N6aRkocws5eQVFWiil/FQASgCuSPVgPygAPooAC0AD4Go1GpddNdZvMIFQAhBgqFwhAXsdTucUABfVrESooOprbQbPRfFEoOh4PAvaq1SEE8m6HZ7A4oQ7dOjMXAjC4slBspD7I7MJCsW4IgVTYWizniyV3HAwGCHIA

Reply:
Oh ok! So nothing too fancy going on. Thanks for the demo! 🙏
I should use sharplab more often...

(I'll update the commit)